### ISSUE TITLE
Changes on duration model

### ISSUE THREAD

**Issue by alexdemartos** [as-ideas/ForwardTacotron#11]
Hi,

I just found the current duration model suffers when synthesizing very long sentences (probably because of the recurrent part of the duration network).

I managed to fix it by feeding prenet's outputs instead of the raw text embeddings, and replacing the current duration model by FastSpeech's (fully convolutional). I also converted the durations to the log domain.

I found these changes not only successfully fix the problems with very long sentences, but also make prosody more natural in multi-speaker settings (specially on silence/pauses modelling).

Hope this is useful somehow.

Best regards.

**Comment by cschaefer26** [comment]
Hi, sounds interesitng, thanks for sharing! I definitely gonna try this again. Do you have a fork I could test it with locally? I would also be interested whether a log for durations really helps, maybe it emphasizes short durations more (given you use l1 or l2 losses).

**Comment by alexdemartos** [comment]
Unfortunately I cannot share the repository since I am working for a private project, but I'll be glad to share ideas.

I am not sure on how much calculating the losses in the log domain helps, I guess not much, but seems reasonable to make the value range more compact (durations can go from 0 more than 100, which correspond to pauses in the audio). The LengthRegulator will convert them back to the linear domain before applying the expansion.

PD: Yes, I also switched to L2 loss for log-durations, forgot to mention.

**Comment by cschaefer26** [comment]
Sounds good! I'll do another training soon with the mentioned changes.

**Comment by m-toman** [comment]
![Screenshot_20200621-113954_Chrome](https://user-images.githubusercontent.com/9433378/85221428-02c26180-b3b4-11ea-831e-064bd1cda1fc.jpg)

Replaced with fastspeech duration model. Subjectively the improvement isn't that huge but noticable in longer sentences. 

Prosody not yet at vanilla taco2 level in my case but close. Overall still quite a bit worse still but the taco2 model I extracted durations from was awful. Perhaps I'll see improvement with forced alignment.

**Comment by cschaefer26** [comment]
Cool thx for sharing. Did you use a vocoder to compare the results?

**Comment by m-toman** [comment]
Yeah when I used our preprocessing it worked with my melgan pretrained model although a bit noisy. I usually do a couple hours finetuning then. Also got a couple wavernn versions, which usually give slightly better results - but melgan is nicer for testing.

Hope I haven't got an error somewhere in the interface because the results are rather noisy. But not noisy enough for something serious ;)
